subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c

```diff
@@ -214,11 +215,205 @@ fs_mgmt_file_upload(struct mgmt_ctxt *ctxt)
 	return fs_mgmt_file_upload_rsp(ctxt, 0, fs_mgmt_ctxt.off);
 }
 
+/**
+ * Command handler: fs size (read)
+ */
+static int
+fs_mgmt_file_size(struct mgmt_ctxt *ctxt)
+{
+	char path[CONFIG_FS_MGMT_PATH_SIZE + 1];
+	CborError err;
+	size_t file_len;
+	int rc;
+
+	const struct cbor_attr_t size_attr[] = {
+		{
+			.attribute = "name",
+			.type = CborAttrTextStringType,
+			.addr.string = path,
+			.len = sizeof(path),
+		},
+		{ 0 },
+	};
+
+	rc = cbor_read_object(&ctxt->it, size_attr);
+
+	if (rc != 0) {
+		return MGMT_ERR_EINVAL;
+	}
+
+	/* Retrieve file size */
+	rc = fs_mgmt_impl_filelen(path, &file_len);
+
+	if (rc != 0) {
+		return rc;
+	}
+
+	/* Encode the response. */
+	err = 0;
+	err |= cbor_encode_text_stringz(&ctxt->encoder, "len");
+	err |= cbor_encode_uint(&ctxt->encoder, file_len);
+	err |= cbor_encode_text_stringz(&ctxt->encoder, "rc");
+	err |= cbor_encode_int(&ctxt->encoder, MGMT_ERR_EOK);
+
+	if (err != 0) {
+		return MGMT_ERR_ENOMEM;
+	}
+
+	return 0;
+}
+
+/**
+ * Command handler: fs checksum (read)
+ */
+static int
+fs_mgmt_file_checksum(struct mgmt_ctxt *ctxt)
+{
+#if CONFIG_FS_MGMT_CHECKSUM_CHUNK_SIZE > 0
+	char path[CONFIG_FS_MGMT_PATH_SIZE + 1];
+	CborError err;
+	size_t file_len;
+	size_t checksum_len;
+	int rc;
+	unsigned long long type;
+	uint32_t checksum;
+
+	const struct cbor_attr_t size_attr[] = {
+		{
+			.attribute = "type",
+			.type = CborAttrUnsignedIntegerType,
+			.addr.uinteger = &type,
+			.nodefault = true
+		},
+		{
+			.attribute = "name",
+			.type = CborAttrTextStringType,
+			.addr.string = path,
+			.len = sizeof(path),
+		},
+		{ 0 },
+	};
+
+	type = FS_MGMT_CHECKSUM_COUNT;
+	rc = cbor_read_object(&ctxt->it, size_attr);
+
+	if (rc != 0 || type >= FS_MGMT_CHECKSUM_COUNT) {
+		return MGMT_ERR_EINVAL;
+	}
+
```

Comment:
I think that it is worth adding `path[ARRAY_SIZE(path) - 1] = '\0'` here to "just-in-case" make sure that null terminator is there.